### ISSUE TITLE
OpenVidu - Firefox can’t establish a connection to the server at AWS

### ISSUE THREAD

**Issue by rida-rech** [OpenVidu/openvidu-tutorials#70]
Hi,
my project with OpenVidu work a good with chrome but I have a problem with firefox, i can't understand why this problem just with Firefox and AWS.(for information
Firefox work with server local)
=> Firefox can’t establish a connection to the server at AWS
**Error firefox :**
`'OpenVidu' initialized OpenVidu.ts:107:12
openvidu-browser version: 2.11.0 OpenVidu.ts:108:12
TURN temp credentials [ZSBPS8:vjig95] Session.ts:1186:24
openvidu-server version: 2.11.0 Session.ts:1195:24
Connecting websocket to URI: wss://ec2-34-253-8-220.eu-west-1.compute.amazonaws.com:4443/openvidu jsonrpcclient.js:138:11
Firefox can’t establish a connection to the server at wss://ec2-34-253-8-220.eu-west-1.compute.amazonaws.com:4443/openvidu. webSocketWithReconnection.js:71:13
Could not connect to wss://ec2-34-253-8-220.eu-west-1.compute.amazonaws.com:4443/openvidu (invoking onerror if defined) 
error { target: WebSocket, isTrusted: true, srcElement: WebSocket, currentTarget: WebSocket, eventPhase: 2, bubbles: false, cancelable: false, returnValue: true, defaultPrevented: false, composed: false, … }
webSocketWithReconnection.js:82:15
--------- ONERROR ----------- jsonrpcclient.js:125:15
Connection closed unexpectecly. Reconnecting... webSocketWithReconnection.js:101:23
reconnectToSameUri (attempt #1, max=2000) webSocketWithReconnection.js:112:15
--------- ONRECONNECTING ----------- jsonrpcclient.js:79:15
Websocket connection lost (reconnecting)`

**Comment by pabloFuente** [comment]
Do OpenVidu Demos (https://demos.openvidu.io/) work fine with that same Firefox installation? If they do, please make sure to meet these network requirements in your instance: https://openvidu.io/docs/deployment/deploying-ubuntu/#server-network-requirements

**Comment by rida-rech** [comment]
Yes OpenVidu Demo (https://demos.openvidu.io/basic-webinar/) work good with my Firefox.

**Comment by rida-rech** [comment]
Hi @pabloFuente pabloFuente  
I deployed OpenVidu Demos on AWS, the same problem I have on my project.
This link OpenVidu Demos on AWS : https://ec2-34-254-171-37.eu-west-1.compute.amazonaws.com/basic-webinar/

Please do you test in your Firefox ?

**Comment by devcarlos** [comment]
I have the same issue with 2.12.0 GitHub Demo, in fact I have opened and issue:

https://github.com/OpenVidu/openvidu/issues/467

Any workaround this ? it seems a lot of months on this issue. 😅

@pabloFuente FYI 🖐️

**Comment by devcarlos** [comment]
FYI

> Yes OpenVidu Demo (https://demos.openvidu.io/basic-webinar/) work good with my Firefox.
> For information : i use default configuration of OpenVidu for template AWS (CF-OpenVidu-latest.yaml ) in my project.

Works good on Firefox as well BUT it seems it is a JAVA demo so I'm a little stuck testing this platform locally with Node.js so far.

**Comment by devcarlos** [comment]
UPDATE:

After test on Chrome I was able to see the Video Session:

<img width="1680" alt="Screen Shot 2020-05-09 at 20 49 06" src="https://user-images.githubusercontent.com/646364/81488425-9d1c7a80-9236-11ea-9f67-5835dbb5a017.png">

So it seems the wss protocol issue is related to Firefox only, will keep looking for a workaround on that browser.

**Comment by devcarlos** [comment]
**UPDATE2: SOLVED ON FIREFOX !**

After open the WSS server URL https://localhost:4443/ on another Firefox Tab (as suggested on some google thread) I was able to JOIN the video session:

<img width="1680" alt="Screen Shot 2020-05-09 at 21 03 22" src="https://user-images.githubusercontent.com/646364/81488599-cd651880-9238-11ea-858f-387472d69eb6.png">

So it the Firefox issue with WSS is a bug: https://groups.google.com/forum/#!topic/netty/sCIB8J6abu8

So I think this issue can be closed now I have found the solution.

🤓👍

**Comment by dahas** [comment]
Hello! Just stumbled upon the same problem. In my case I had to exclude the AWS url from Bitdefender. To do so I executed the AWS url in Firefox and when Bitdefender blocks the url I clicked on "I know the risk" (or what ever it said). Then it worked.
